presto-main/src/test/java/com/facebook/presto/operator/TestDynamicFilterSourceOperator.java

```diff
@@ -255,6 +272,25 @@ public void testCollectTooManyRows()
         assertEquals(partitions.build(), ImmutableList.of(TupleDomain.all()));
     }
 
+    @Test
+    public void testCollectTooManyRowIntoRange()
+    {
+        int maxRowCount = getDynamicFilteringMaxPerDriverRowCount(pipelineContext.getSession());
+        Page largePage = createSequencePage(ImmutableList.of(BIGINT), maxRowCount + 1);
+
+        OperatorFactory operatorFactory = createOperatorFactory(
+                ImmutableList.of(channel(0, BIGINT)),
+                getDynamicFilteringMaxPerDriverRowCount(pipelineContext.getSession()),
+                getDynamicFilteringMaxPerDriverSize(pipelineContext.getSession()),
+                largePage.getPositionCount());
+        verifyPassThrough(createOperator(operatorFactory), ImmutableList.of(BIGINT), largePage);
+        operatorFactory.noMoreOperators();
+        assertEquals(
+                partitions.build(),
+                ImmutableList.of(TupleDomain.withColumnDomains(ImmutableMap.of(
+                        "0", Domain.create(ValueSet.ofRanges(Range.range(BIGINT, 0L, true, (long) maxRowCount, true)), false)))));
+    }
+
```

Comment:
Can we also port other newly added tests in this class?

Reply:
Sounds good, ported the latest `TestDynamicFilterSourceOperator`.